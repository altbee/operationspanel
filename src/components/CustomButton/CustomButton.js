import React from "react";
import { Button } from "react-bootstrap";
import cx from "classnames";
import PropTypes from "prop-types";

export const CustomButton = props => {
  const { fill, simple, pullRight, round, block, ...rest } = props;

  const btnClasses = cx({
    "btn-fill": fill,
    "btn-simple": simple,
    "pull-right": pullRight,
    "btn-block": block,
    "btn-round": round
  });

  return <Button className={btnClasses} {...rest} />;
};

CustomButton.propTypes = {
  fill: PropTypes.bool.isRequired,
  simple: PropTypes.bool.isRequired,
  pullRight: PropTypes.bool.isRequired,
  block: PropTypes.bool.isRequired,
  round: PropTypes.bool.isRequired
};

export default CustomButton;
