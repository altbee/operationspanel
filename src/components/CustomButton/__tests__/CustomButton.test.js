import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { CustomButton } from "../CustomButton";

configure({ adapter: new Adapter() });

describe("<CustomButton />", () => {
  const requiredProps = {
    fill: true,
    simple: true,
    pullRight: true,
    round: true,
    block: true
  };

  const rest = {
    disabled: true
  };

  const rendered = mount(<CustomButton {...requiredProps} />);

  it("renders with expected required props", () => {
    expect(rendered.find("button").hasClass("btn-fill")).toEqual(true);
    expect(rendered.find("button").hasClass("btn-simple")).toEqual(true);
    expect(rendered.find("button").hasClass("pull-right")).toEqual(true);
    expect(rendered.find("button").hasClass("btn-block")).toEqual(true);
    expect(rendered.find("button").hasClass("btn-round")).toEqual(true);
  });

  it("does not apply classes to child if props are false", () => {
    rendered.setProps({ simple: false, round: false });
    expect(rendered.find("button").hasClass("btn-simple")).toEqual(false);
    expect(rendered.find("button").hasClass("btn-round")).toEqual(false);
  });

  it("passes down the rest object props to child", () => {
    const wrapper = mount(<CustomButton {...requiredProps} {...rest} />);
    expect(wrapper.find("button[disabled=true]").exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    const tree = mount(<CustomButton {...requiredProps} />);
    expect(tree).toMatchSnapshot();
  });
});
