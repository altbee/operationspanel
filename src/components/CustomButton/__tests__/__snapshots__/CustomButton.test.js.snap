// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> matches the snapshot 1`] = `
<CustomButton
  block={true}
  fill={true}
  pullRight={true}
  round={true}
  simple={true}
>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="btn-fill btn-simple pull-right btn-block btn-round"
    disabled={false}
  >
    <button
      className="btn-fill btn-simple pull-right btn-block btn-round btn btn-default"
      disabled={false}
      type="button"
    />
  </Button>
</CustomButton>
`;
